### PR TITLE
Update endpoints for dynamic column fetching

### PR DESCRIPTION
**Pull request by brittanylau**
### Notion ticket link

[Dynamically generate table rows & columns based on JSON fields](https://www.notion.so/uwblueprintexecs/9cb454262ed84c6299c0682c23f92ec5?v=1076bd30711a4a01838becee05f8b5a3&p=a26dba9883ad47d3a0c866dcaa3c3710)

### Implementation description

- updated the FamilySerializer to return the family's parent
- created a FieldListSerializer with `to_representation` overridden to return fields grouped by `role`, ordered by `order`
- updated tests

### Steps to test

1. `python manage.py test`

### What should reviewers focus on?

- `to_representation` implementation

### Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] I have run the linter
- [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: registration/tests/serializers/test_field_list.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            role=Field.GUEST,
+            name="Relation to Family",
+            question="How are you related?",
+            question_type=Field.TEXT,
+            is_default=True,
+            order=1,
+        )
+
+    def test_field_list_serializer(self):
+        serializer = FieldListSerializer(child=FieldSerializer(), data=Field.objects)
```
Comment (Tayef-Shah) [inline registration/tests/serializers/test_field_list.py:47]: kinda confused why the first parameter is called `child`, does it mean like child serializer?

**Review by Tayef-Shah — APPROVED** [review]
LGTM

File: registration/tests/serializers/test_field_list.py (hunk 1 above)
Comment (brittanylau) [reply registration/tests/serializers/test_field_list.py:47]: this is a [DRF thing](https://www.django-rest-framework.org/api-guide/serializers/#listserializer) - the FieldSerializer is a child of the FieldListSerializer!

i think we're just really good at naming our models after keywords 😅

Comment (Tayef-Shah) [reply registration/tests/serializers/test_field_list.py:47]: yeah that's what I meant lol. Thanks for clarifying! That's a kewl feature

File: registration/models.py [hunk 2]
```diff
@@ -70,6 +70,10 @@ def __str__(self):
     class Meta:
         verbose_name_plural = "students"
 
+    def clean(self):
+        validate_student(self)
+        return super().clean()
```
Comment (annoyingpr) [inline registration/models.py]: What does `super().clean()` do?

Comment (brittanylau) [reply registration/models.py]: it calls the OG `clean` method, which will ensure all the other fields are also cleaned 🧼

**Pull request merged**